### DISCUSSION TITLE
A Device Type template should allow the user to define LAG components

### DISCUSSION THREAD

**Discussion by tim-fiola** [nautobot/nautobot#544] [Ideas]
In the Device Type template, the user can currently add both LAG interfaces and physical interfaces (for example, ae0 and xe-0/0/[0-9].

However, the user is not allowed to assign component interfaces to the LAG in that template. 
![Screen Shot 2021-06-08 at 1 02 29 PM](https://user-images.githubusercontent.com/13647153/121242456-d10f0880-c859-11eb-835f-b2bfbf70f99a.png)

Instead, the user must go to each instantiation of that device type and assign the component interfaces to the LAG.

If this is a true Device Type template, why would we not allow the user to define this default LAG component assignment?

![Screen Shot 2021-06-08 at 1 03 26 PM](https://user-images.githubusercontent.com/13647153/121242553-ef750400-c859-11eb-9b52-18c5d70161ae.png)

**Comment by itdependsnetworks** [comment]
How it operates currently is inline with my personal expectation. Physical devices rarely (if ever?) enforce LAGs, and the device type is meant to model a physical device. The fact that logical interfaces can be defined in the template, I would consider happenstance, and potentially even something that is not supported in the future. 

This is one outsiders view, but in passing, thought I would articulate it :)

**Comment by smk4664** [comment]
So, as far has Device Types being source's of truth, like the https://github.com/netbox-community/devicetype-library, I don't think LAG's pre-filled with Interfaces to be in the Device Type.

BUT, If Manager Mary declares that all of the Core switches should have LAG ae0 with Interfaces xe-0/0/1 and xe-1/0/1, why not make that a template...
